lib/token-request.js

```diff
@@ -559,7 +566,10 @@ TokenRequest.prototype.getTokenWithCertificate = function(certificate, thumbprin
   oauthParameters[OAuth2Parameters.CLIENT_ASSERTION_TYPE] = OAuth2GrantType.JWT_BEARER;
   oauthParameters[OAuth2Parameters.CLIENT_ASSERTION] = jwt;
 
-  this._getTokenWithCacheWrapper(callback, function(getTokenCompleteCallback) {
+  this._getTokenWithCacheWrapper(callback, function (err, getTokenCompleteCallback) {
+    if (err) {
```

Comment:
this will break some of the flow. The logic here is, we look into the cache first and try to use the refresh token, if it doesn't work, we start the new flow.